clang/lib/Driver/Driver.cpp

```diff
@@ -7851,8 +7867,17 @@ Driver::getOffloadArchs(Compilation &C, const llvm::opt::DerivedArgList &Args,
       Archs.insert(CudaArchToString(CudaArch::HIPDefault));
     else if (Kind == Action::OFK_OpenMP)
       Archs.insert(StringRef());
-    else if (Kind == Action::OFK_SYCL)
-      Archs.insert(StringRef());
+    else if (Kind == Action::OFK_SYCL) {
+      // For SYCL offloading, we need to check the triple for NVPTX or AMDGPU
+      // and set the default arch accordingly
+      if (TC->getTriple().isNVPTX())
+        Archs.insert(CudaArchToString(CudaArch::SM_50));
+      else if (TC->getTriple().isAMDGPU())
+        C.getDriver().Diag(clang::diag::err_drv_sycl_missing_amdgpu_arch)
```

Comment:
>> we need to check the triple for NVPTX or AMDGPU and set the default arch accordingly
Why are we just checking if the triple is AMD and just emitting an error? Shouldn't we set the arch first?

Reply:
The error is that there is a missing arch setting.  The AMD path requires an arch to be supplied (unlike nvptx where we set a default arch).

Reply:
I will update the description to be more clear of our intentions in regards to AMD

Reply:
I have updated the comment.